### PR TITLE
tools: automate GitHub release creation

### PR DESCRIPTION
**Pull request by MylesBorins**
This extends the `publish.yml` workflow to automate the creation of a GitHub release with automated release notes

Signed-off-by: Myles Borins <mylesborins@github.com>

#### Summary

This further automates the release process publishing to npm + creating a GitHub release from any pushed tag

#### Release Note

* update `publish.yml` to automate GitHub releases

#### Documentation

N/A

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 name: publish
 
 on:
-  release:
-    types: [published]
+  push:
+    tags:
+      - 'v*'
```
Comment (MylesBorins) [inline .github/workflows/publish.yml:6]: With this change you can locally do

```
$ npm version patch|minor|major
$ git push origin main --follow-tags
```

and GitHub will do the rest 👍🏼

**Review by MylesBorins — COMMENTED** [review]
i also have a manually triggered workflow that can be used to create the tag, can't propose that in a follow up PR

File: .github/workflows/publish.yml (hunk 1 above)
Comment (bdehamer) [reply .github/workflows/publish.yml:6]: @MylesBorins can you update the publish section in the readme with this bit

**Review by bdehamer — APPROVED** [review]
I love this! Thanks so much @MylesBorins 

I haven't used the `gh release` command before. Any tips on how we should format our commit messages so that we get something useful/readable out of this?

**Comment by feelepxyz** [comment]
Nice! 

> Any tips on how we should format our commit messages so that we get something useful/readable out of this?

@bdehamer looks like this can be configured somewhat using a config file: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#example-configurations

I'm still curious if we should adopt something like [changesets](https://github.com/changesets/changesets) to generate the changelogs from structured files to you can introduce with each PR, automatically running `npm version x` for you etc. It also comes with an action to create a "release" PR with all changes for the next release rolled up, based on the changesets you've created between the last release and default branch head.

Saying that, changesets feels somewhat heavy though and I like the simplify of relying on github building blocks as much as possible.

**Comment by MylesBorins** [comment]
Looking at the current release notes it looks like you are already using the automatic release note generation feature from GitHub releases, the GH CLI uses the same feature.

It uses the title of the PR not individual commits

File: .github/workflows/publish.yml (hunk 1 above)
Comment (MylesBorins) [reply .github/workflows/publish.yml:6]: done PTAL

**Comment by feelepxyz** [comment]
Closing this as we're now using changesets which creates the github release for us amongst other things: https://github.com/sigstore/sigstore-js/pull/339

**Pull request closed without merging**